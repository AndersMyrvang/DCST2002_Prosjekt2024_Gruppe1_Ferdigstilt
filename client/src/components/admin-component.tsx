import React, { useState, useEffect } from 'react';
import { User } from '../types';

const AdminPanel = () => {
  const [users, setUsers] = useState<User[]>([]);

  const fetchUsers = () => {
    fetch('/api/users')
      .then((response) => response.json())
      .then((data) => setUsers(data));
  };

  useEffect(() => {
    fetchUsers();

    const interval = setInterval(fetchUsers, 60000);

    return () => clearInterval(interval);
  }, []);

  // Sjekker om en bruker er online basert på last_login og last_logout
  const isUserOnline = (user: User) => {
    if (
      user.last_login &&
      (!user.last_logout || new Date(user.last_login) > new Date(user.last_logout))
    ) {
      return true; 
    }
    return false;
  };

  // Funksjon for å endre admin-status
  const toggleAdminStatus = (userId: number, isAdmin: boolean) => {
    fetch(`/api/users/${userId}/set-admin`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ isAdmin: !isAdmin }),
    })
      .then((response) => {
        if (response.ok) {
          setUsers(
            users.map((user) => (user.user_id === userId ? { ...user, is_admin: !isAdmin } : user)),
          );
        } else {
          console.error('Feil ved oppdatering av admin-status');
        }
      })
      .catch((error) => console.error('Feil ved toggleAdminStatus:', error));
  };

  return (
    <div className="admin-panel">
      <table>
        <thead>
          <tr>
            <th>User ID</th>
            <th>Username</th>
            <th>Email</th>
            <th>Member Since</th>
            <th>Online Status</th>
            <th>Admin</th>
          </tr>
        </thead>
        <tbody>
          {users.map((user) => (
            <tr key={user.user_id}>
              <td>{user.user_id}</td>
              <td>{user.username}</td>
              <td>{user.email}</td>
              <td>{new Date(user.first_login).toLocaleDateString()}</td>
              <td>
                <span className={isUserOnline(user) ? 'online-dot' : 'offline-dot'}></span>
              </td>
              <td>
                <input
                  type="checkbox"
                  checked={user.is_admin}
                  onChange={() => toggleAdminStatus(user.user_id!, user.is_admin)}
                />
              </td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default AdminPanel;
