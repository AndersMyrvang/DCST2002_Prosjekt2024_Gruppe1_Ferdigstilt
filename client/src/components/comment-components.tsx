import * as React from 'react';
import { Component } from 'react-simplified';
import { Alert, Card, Row, Column, Form, Button, Card2, Card3 } from '../widgets';
import { NavLink } from 'react-router-dom';
import commentService from '../services/comment-service';
import userService from '../services/user-service';
import { Comment, User } from '../types';
import { createHashHistory } from 'history';
import { AiFillLike, AiFillDislike } from 'react-icons/ai';
import { FaTrashAlt } from 'react-icons/fa';
import { BiSolidPencil } from 'react-icons/bi';

const history = createHashHistory();

interface CommentSectionProps {
  page_id: number;
  onNewComment: () => void;
}

//Komponent for ny kommentar
export class NewComment extends Component<CommentSectionProps> {
  comment: Comment = {
    comment_id: 0,
    user_id: 0,
    content: '',
    created_at: new Date(),
    updated_at: new Date(),
    page_id: 0,
    likes: 0,
  };

  render() {
    return (
      <>
        <Card title="New Comment">
          <Column>
            <Form.Input
              type="text"
              aria-label={`Comment input`}
              value={this.comment.content}
              onChange={(event) => (this.comment.content = event.currentTarget.value)}
              rows={3}
            />
          </Column>
          <Column>
            <Button.Success
              onClick={() => {
                this.NewComment(this.comment.user_id, this.comment.content, this.props.page_id);
                this.comment.content = '';
              }}
            >
              {' '}
              SEND{' '}
            </Button.Success>
          </Column>
        </Card>
      </>
    );
  }
  mounted() {
    const currentUserId = localStorage.getItem('user_id');
    if (currentUserId != null) this.comment.user_id = parseInt(currentUserId, 10);
  }
  NewComment(user_id: number, content: string, page_id: number) {
    commentService
      .create(user_id, content, page_id)
      .then((comment) => {
        this.props.onNewComment();
      })
      .catch((error) => {
        console.error('Error creating comment:', error);
      });
  }
}

//Komponent for kommentarseksjonen (visning av kommentarer p√• en spesifikk side)
export class CommentSection extends Component<CommentSectionProps> {
  comment: Comment[] = [];
  user: User = {
    user_id: 0,
    google_id: '',
    username: '',
    email: '',
    profile_image_url: '',
    first_login: new Date(),
    last_login: new Date(),
    last_logout: new Date(),
    is_admin: false,
  };
  updatingComment: number | null = null;
  updatedContent: string = '';

  render() {
    return (
      <>
        <Card title="Comments">
          {this.user.user_id != 0 ? (
            <NewComment page_id={this.props.page_id} onNewComment={() => this.mounted()} />
          ) : null}
          {this.comment.map((comment) => (
            <Row key={new Date(comment.created_at).getTime()}>
              <Column>
                {' '}
                <img
                  src={`${comment.profile_image_url}`}
                  alt={`Bilde fra ${comment.profile_image_url}`}
                  width={'45px'}
                  style={{
                    borderRadius: '50%',
                    objectFit: 'cover',
                  }}
                />
              </Column>
              <Column>
                <b>{comment.username}</b>
              </Column>
              <Column>
                {this.updatingComment == comment.comment_id ? (
                  <Form.Input
                    type="text"
                    value={this.updatedContent}
                    onChange={(event) => (this.updatedContent = event.currentTarget.value)}
                  />
                ) : (
                  comment.content
                )}
              </Column>

              <Column>
                {this.user.user_id == comment.user_id || this.user.is_admin == true ? (
                  <Button.Danger
                    onClick={() => {
                      this.deleteComment(comment.comment_id);
                      this.mounted();
                    }}
                  >
                    <FaTrashAlt />
                  </Button.Danger>
                ) : null}
              </Column>
              <Column>
                {this.user.user_id == comment.user_id || this.user.is_admin == true ? (
                  <>
                    {this.updatingComment == comment.comment_id ? (
                      <>
                        <Button.Success
                          onClick={() => {
                            this.updateComment(comment.comment_id, this.updatedContent);
                            this.updatingComment = null;
                            this.mounted();
                          }}
                        >
                          Lagre
                        </Button.Success>
                      </>
                    ) : (
                      <>
                        <Button.Light
                          onClick={() => {
                            this.updatingComment = comment.comment_id;
                            this.updatedContent = comment.content;
                          }}
                        >
                          <BiSolidPencil />
                        </Button.Light>
                      </>
                    )}
                  </>
                ) : null}
              </Column>
            </Row>
          ))}
        </Card>
      </>
    );
  }

  mounted() {
    const current_userId = localStorage.getItem('user_id');
    if (current_userId != null) {
      userService
        .getUserById(parseInt(current_userId, 10))
        .then((user: User) => {
          this.user = user;
        })
        .catch((error) => {
          console.error('Error getting user: ', error);
          Alert.danger('Error getting user: ' + error.message);
        });
    }

    commentService
      .listComments(this.props.page_id)
      .then(async (comments: Comment[]) => {
        this.comment = comments;
      })
      .catch((error) => {
        console.error('Error fetching comments:', error);
        Alert.danger('Error getting comments: ' + error.message);
      });
  }

  deleteComment(comment_id: number) {
    commentService.deleteComment(comment_id);
  }
  updateComment(comment_id: number, content: string) {
    commentService.update(comment_id, content);
  }
}

export default CommentSection;
