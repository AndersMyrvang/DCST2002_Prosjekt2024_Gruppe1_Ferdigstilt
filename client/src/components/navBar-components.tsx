import React, { useState, useRef, useEffect } from 'react';
import SearchBar from './searchBar-component';
import { FaHouse } from 'react-icons/fa6';
import { User } from '../types';

interface HamburgerProps {
  user: User | null;
}

function DarkModeButton() {
  const [darkMode, setDarkMode] = useState(() => {
    const savedMode = localStorage.getItem('darkMode');
    return savedMode === 'true';
  });

  const toggleDarkMode = () => {
    const newMode = !darkMode;
    setDarkMode(newMode);
    document.body.classList.toggle('dark-mode', newMode);
    localStorage.setItem('darkMode', newMode.toString());
  };

  useEffect(() => {
    if (darkMode) {
      document.body.classList.add('dark-mode');
    }
  }, [darkMode]);

  return (
    <button className="dark-mode-button" onClick={toggleDarkMode} aria-label="Toggle Dark Mode">
      <i
        className={darkMode ? 'fas fa-sun' : 'fas fa-moon'}
        style={{ fontSize: '24px', color: darkMode ? '#FFD700' : '#555' }}
      />
    </button>
  );
}

const Hamburger: React.FC<HamburgerProps> = ({ user }) => {
  const [burgerClass, setBurgerClass] = useState('burger-bar unclicked');
  const [menuClass, setMenuClass] = useState('menu hidden');
  const [isMenuClicked, setIsMenuClicked] = useState(false);

  const [isLoggedIn, setIsLoggedIn] = useState(() => localStorage.getItem('isLoggedIn') === 'true');

  const [userName, setUserName] = useState(localStorage.getItem('user') || '');
  const menuRef = useRef<HTMLDivElement>(null);

  const updateMenu = () => {
    setIsMenuClicked((prev) => !prev);
    setBurgerClass(isMenuClicked ? 'burger-bar unclicked' : 'burger-bar clicked');
    setMenuClass(isMenuClicked ? 'menu hidden' : 'menu visible');
  };

  const handleClickOutside = (event: MouseEvent) => {
    if (menuRef.current && !menuRef.current.contains(event.target as Node)) {
      closeMenu();
    }
  };

  const closeMenu = () => {
    setBurgerClass('burger-bar unclicked');
    setMenuClass('menu hidden');
    setIsMenuClicked(false);
  };

  const handleSearch = (query: string) => {
    fetch(`/api/search?q=${query}`)
      .then((response) => response.json())
      .then((data) => {})
      .catch((error) => {
        console.error('Feil ved søk:', error);
      });
  };

  const handleLogout = () => {
    fetch('/api/logout', { method: 'GET', credentials: 'include' })
      .then((response) => {
        if (response.ok) {
          setIsLoggedIn(false);
          setUserName('');
          localStorage.removeItem('isLoggedIn');
          localStorage.removeItem('user');
          localStorage.removeItem('user_id');
          window.location.href = '/';
        } else {
          console.error('Logout failed');
        }
      })
      .catch((error) => console.error('Logout error:', error));
  };

  useEffect(() => {
    document.addEventListener('mousedown', handleClickOutside);

    return () => {
      document.removeEventListener('mousedown', handleClickOutside);
    };
  }, []);

  // Sjekker om bruker er pålogget
  useEffect(() => {
    fetch('/api/current_user')
      .then((response) => response.json())
      .then((data) => {
        if (data && data.username) {
          setIsLoggedIn(true);
          setUserName(data.username);
          localStorage.setItem('isLoggedIn', 'true');
          localStorage.setItem('user', data.username);
          localStorage.setItem('user_id', data.user_id);
        } else {
          setIsLoggedIn(false);
          localStorage.setItem('isLoggedIn', 'false');
        }
      })
      .catch(() => {
        setIsLoggedIn(false);
        localStorage.setItem('isLoggedIn', 'false');
      });
  }, []);

  return (
    <div style={{ width: '100%' }} ref={menuRef}>
      <nav>
        <div className="house">
          <a href="/">
            <FaHouse />
          </a>
        </div>
        <div className="burger-menu" onClick={updateMenu} aria-label="Toggle Menu">
          <div className={burgerClass}></div>
          <div className={burgerClass}></div>
          <div className={burgerClass}></div>
        </div>

        <SearchBar onSearch={handleSearch} />

        <DarkModeButton />
      </nav>

      <div className={menuClass}>
        <ul className="menu-links" onClick={closeMenu} role="menu">
          <li>
            <a href="/">Home</a>
          </li>
          <li>
            <a href="#players">Players</a>
          </li>
          <li>
            <a href="#teams">Teams</a>
          </li>
          <li>
            <a href="#leagues">Leagues</a>
          </li>
          <li>
            <a href="#tags">Tags</a>
          </li>

          {!isLoggedIn ? (
            <li>
              <a href="https://accounts.google.com/signup/v2/createaccount">Register </a> /
              <a href="/auth/google"> Log in with Google </a>
            </li>
          ) : (
            <>
              <li>
                <a href="#/profile">My profile </a> /{' '}
                <a href="#" onClick={handleLogout}>
                  Log out
                </a>
              </li>
            </>
          )}
        </ul>
      </div>
    </div>
  );
};

export default Hamburger;
