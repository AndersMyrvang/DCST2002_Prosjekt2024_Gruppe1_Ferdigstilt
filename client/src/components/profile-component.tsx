import React, { useEffect, useState } from 'react';
import AdminPanel from './admin-component';
import { UserProfile } from '../types';

export const useUser = () => {
  const [user, setUser] = useState<UserProfile | null>(null);

  useEffect(() => {
    fetch('/api/current_user')
      .then((response) => response.json())
      .then((data) => {
        if (data) {
          setUser({
            username: data.username,
            email: data.email,
            photo: data.photo,
            firstLogin: new Date(data.firstLogin).toLocaleDateString(),
            lastLogin: new Date(data.lastLogin).toLocaleDateString(),
            is_admin: Boolean(data.is_admin),
            user_id: data.user_id,
          });
        }
      })
      .catch((error) => console.error('Error fetching user data:', error));
  }, []);

  return user;
};

const Profile = () => {
  const user = useUser();
  if (!user) {
    return <div>Laster...</div>;
  }

  return (
    <div className="profile-page">
      <div className="profile-card">
        <div className="profile-card-content">
          <div className="profile-header">
            <img src={user.photo} alt="Profilbilde" className="profile-img" />
          </div>
          <div className="profile-content">
            <h2 className="profile-name">{user.username}</h2>
            <p className="profile-info">Member since {user.firstLogin}</p>
            <p className="profile-details">
              <span>Last visited {user.lastLogin}</span>
              <br />
              <br />
              <span>{user.email}</span>
            </p>
          </div>
        </div>
      </div>

      {user.is_admin && (
        <div className="admin-panel-container">
          <h2>Admin Panel</h2>
          <AdminPanel />
        </div>
      )}
    </div>
  );
};

export default Profile;
