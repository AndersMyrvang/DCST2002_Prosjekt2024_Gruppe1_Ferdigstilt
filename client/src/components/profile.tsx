import React, { useEffect, useState } from 'react';

// Info om bruker som skal vises
interface UserProfile {
  displayName: string;
  email: string;
  photo: string;
}

// Lager komponent
const Profile = () => {
  const [user, setUser] = useState<UserProfile | null>(null);

  useEffect(() => {
    fetch('/api/current_user')
      .then((response) => response.json())
      .then((data) => {
        if (data) {
          setUser({
            displayName: data.displayName,
            email: data.emails && data.emails[0]?.value,
            photo: data.photos && data.photos[0]?.value,
          });
        }
      });
  }, []);

  // Dersom det ikke finnes eller laster inn tregt
  if (!user) {
    return <div>Laster...</div>;
  }

  // Viser info
  return (
    <div style={{ textAlign: 'center', marginTop: '20px' }}>
      <img
        src={user.photo}
        alt="Profilbilde"
        style={{ borderRadius: '50%', width: '150px', height: '150px', marginBottom: '20px' }}
      />
      <h2>{user.displayName}</h2>
      <p>{user.email}</p>
      <h3>"Vise lister eller noe annet her?"</h3>
    </div>
  );
};

export default Profile;
