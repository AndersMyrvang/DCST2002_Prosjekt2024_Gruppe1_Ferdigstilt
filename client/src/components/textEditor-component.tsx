import { Editor } from '@tinymce/tinymce-react';
import React from 'react';

interface TextEditorProps {
  initialContent: string;
  onContentChange: (content: string) => void;
}

//Komponent for skrivefelt i detaljsidene av spiller/league/team
export function TextEditor({ initialContent, onContentChange }: TextEditorProps) {
  return (
    <Editor
      value={initialContent}
      onEditorChange={(content) => onContentChange(content)}
      init={{
        height: 500,
        menubar: false,
        plugins: [
          'advlist',
          'autolink',
          'lists',
          'link',
          'image',
          'charmap',
          'print',
          'preview',
          'anchor',
          'searchreplace',
          'visualblocks',
          'code',
          'fullscreen',
          'insertdatetime',
          'media',
          'table',
          'paste',
          'code',
          'help',
          'wordcount',
        ],
        toolbar:
          'undo redo | formatselect | bold italic backcolor | alignleft aligncenter alignright alignjustify | bullist numlist outdent indent | removeformat | help',
        setup: function (editor: any) {
          editor.on('Change', function () {
            onContentChange(editor.getContent());
          });
          editor.on('init', function () {
            const toolbar = document.querySelector('.tox-editor-header');
            if (toolbar) {
              (toolbar as HTMLElement).style.zIndex = '0';
            }
          });
        },
      }}
    />
  );
}
