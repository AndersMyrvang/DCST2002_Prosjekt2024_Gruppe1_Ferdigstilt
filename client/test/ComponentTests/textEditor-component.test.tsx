import * as React from 'react';
import { Component } from 'react-simplified';
import { shallow, mount } from 'enzyme';
import { TextEditor } from '../../src/components/textEditor-component';
import { Editor } from '@tinymce/tinymce-react';

describe('TextEditor component tests', () => {
  const mockOnContentChange = jest.fn();
  const initialContent = 'Initial content';

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('TextEditor component renders correctly with initial content', () => {
    const wrapper = shallow(
      <TextEditor initialContent={initialContent} onContentChange={mockOnContentChange} />,
    );
    expect(wrapper.find(Editor).prop('value')).toBe(initialContent);
    expect(wrapper).toMatchSnapshot();
  });

  test('TextEditor calls onContentChange when editor content changes', () => {
    const wrapper = shallow(
      <TextEditor initialContent={initialContent} onContentChange={mockOnContentChange} />,
    );

    const mockEditorInstance = {};
    const onEditorChange = wrapper.find(Editor).prop('onEditorChange');
    if (onEditorChange) {
      onEditorChange('Updated content', mockEditorInstance);
    }

    expect(mockOnContentChange).toHaveBeenCalledWith('Updated content');
  });

  test('TextEditor correctly updates when initialContent changes', () => {
    const wrapper = shallow(
      <TextEditor initialContent="First content" onContentChange={mockOnContentChange} />,
    );
    expect(wrapper.find(Editor).prop('value')).toBe('First content');

    wrapper.setProps({ initialContent: 'Updated content' });
    wrapper.update();
    expect(wrapper.find(Editor).prop('value')).toBe('Updated content');
  });

  test('TextEditor setup function initializes correctly', () => {
    const wrapper = shallow(
      <TextEditor initialContent={initialContent} onContentChange={mockOnContentChange} />,
    );

    const initConfig = wrapper.find(Editor).prop('init');
    expect(initConfig).toMatchObject({
      height: 500,
      menubar: false,
      plugins: expect.arrayContaining(['advlist', 'autolink', 'lists', 'link']),
      toolbar: expect.stringContaining('undo redo | formatselect | bold italic backcolor'),
    });
  });

  test('TextEditor snapshot matches with initial content', () => {
    const wrapper = shallow(
      <TextEditor initialContent={initialContent} onContentChange={mockOnContentChange} />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('TextEditor renders correctly with empty initial content', () => {
    const wrapper = shallow(<TextEditor initialContent="" onContentChange={mockOnContentChange} />);
    expect(wrapper.find(Editor).prop('value')).toBe('');
    expect(wrapper).toMatchSnapshot();
  });

  test('TextEditor does not call onContentChange during initialization', () => {
    const wrapper = shallow(
      <TextEditor initialContent={initialContent} onContentChange={mockOnContentChange} />,
    );

    expect(mockOnContentChange).not.toHaveBeenCalled();
  });

  test('TextEditor correctly handles very large content', () => {
    const largeContent = 'a'.repeat(10000);
    const wrapper = shallow(
      <TextEditor initialContent={largeContent} onContentChange={mockOnContentChange} />,
    );
    expect(wrapper.find(Editor).prop('value')).toBe(largeContent);

    wrapper.find(Editor).prop('onEditorChange')!(largeContent, {});
    expect(mockOnContentChange).toHaveBeenCalledWith(largeContent);
  });

  test('TextEditor component renders correctly with initial content', () => {
    const wrapper = shallow(
      <TextEditor initialContent={initialContent} onContentChange={mockOnContentChange} />,
    );
    expect(wrapper.find(Editor).prop('value')).toBe(initialContent);
    expect(wrapper).toMatchSnapshot();
  });

  test('TextEditor calls onContentChange when editor content changes', () => {
    const wrapper = shallow(
      <TextEditor initialContent={initialContent} onContentChange={mockOnContentChange} />,
    );

    const mockEditorInstance = {};
    const onEditorChange = wrapper.find(Editor).prop('onEditorChange');
    if (onEditorChange) {
      onEditorChange('Updated content', mockEditorInstance);
    }

    expect(mockOnContentChange).toHaveBeenCalledWith('Updated content');
  });

  test('TextEditor triggers Change event and calls onContentChange', () => {
    const wrapper = shallow(
      <TextEditor initialContent={initialContent} onContentChange={mockOnContentChange} />,
    );

    const mockEditorInstance = {
      getContent: jest.fn().mockReturnValue('Updated content'),
      on: jest.fn(),
    };

    const init = wrapper.find(Editor).prop('init');
    if (init) {
      init.setup(mockEditorInstance);
    }

    mockEditorInstance.on.mock.calls.forEach((args) => {
      if (args[0] === 'Change') {
        args[1]();
      }
    });

    expect(mockOnContentChange).toHaveBeenCalledWith('Updated content');
    expect(mockEditorInstance.getContent).toHaveBeenCalled();
  });

  test('TextEditor renders correctly with empty initial content', () => {
    const wrapper = shallow(<TextEditor initialContent="" onContentChange={mockOnContentChange} />);
    expect(wrapper.find(Editor).prop('value')).toBe('');
    expect(wrapper).toMatchSnapshot();
  });
});
