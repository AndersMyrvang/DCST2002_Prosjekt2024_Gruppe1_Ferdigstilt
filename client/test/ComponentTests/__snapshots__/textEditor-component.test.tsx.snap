// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextEditor component tests TextEditor component renders correctly with initial content 1`] = `
<Editor
  cloudChannel="7"
  init={
    {
      "height": 500,
      "menubar": false,
      "plugins": [
        "advlist",
        "autolink",
        "lists",
        "link",
        "image",
        "charmap",
        "print",
        "preview",
        "anchor",
        "searchreplace",
        "visualblocks",
        "code",
        "fullscreen",
        "insertdatetime",
        "media",
        "table",
        "paste",
        "code",
        "help",
        "wordcount",
      ],
      "setup": [Function],
      "toolbar": "undo redo | formatselect | bold italic backcolor | alignleft aligncenter alignright alignjustify | bullist numlist outdent indent | removeformat | help",
    }
  }
  onEditorChange={[Function]}
  value="Initial content"
/>
`;

exports[`TextEditor component tests TextEditor component renders correctly with initial content 2`] = `
<Editor
  cloudChannel="7"
  init={
    {
      "height": 500,
      "menubar": false,
      "plugins": [
        "advlist",
        "autolink",
        "lists",
        "link",
        "image",
        "charmap",
        "print",
        "preview",
        "anchor",
        "searchreplace",
        "visualblocks",
        "code",
        "fullscreen",
        "insertdatetime",
        "media",
        "table",
        "paste",
        "code",
        "help",
        "wordcount",
      ],
      "setup": [Function],
      "toolbar": "undo redo | formatselect | bold italic backcolor | alignleft aligncenter alignright alignjustify | bullist numlist outdent indent | removeformat | help",
    }
  }
  onEditorChange={[Function]}
  value="Initial content"
/>
`;

exports[`TextEditor component tests TextEditor renders correctly with empty initial content 1`] = `
<Editor
  cloudChannel="7"
  init={
    {
      "height": 500,
      "menubar": false,
      "plugins": [
        "advlist",
        "autolink",
        "lists",
        "link",
        "image",
        "charmap",
        "print",
        "preview",
        "anchor",
        "searchreplace",
        "visualblocks",
        "code",
        "fullscreen",
        "insertdatetime",
        "media",
        "table",
        "paste",
        "code",
        "help",
        "wordcount",
      ],
      "setup": [Function],
      "toolbar": "undo redo | formatselect | bold italic backcolor | alignleft aligncenter alignright alignjustify | bullist numlist outdent indent | removeformat | help",
    }
  }
  onEditorChange={[Function]}
  value=""
/>
`;

exports[`TextEditor component tests TextEditor renders correctly with empty initial content 2`] = `
<Editor
  cloudChannel="7"
  init={
    {
      "height": 500,
      "menubar": false,
      "plugins": [
        "advlist",
        "autolink",
        "lists",
        "link",
        "image",
        "charmap",
        "print",
        "preview",
        "anchor",
        "searchreplace",
        "visualblocks",
        "code",
        "fullscreen",
        "insertdatetime",
        "media",
        "table",
        "paste",
        "code",
        "help",
        "wordcount",
      ],
      "setup": [Function],
      "toolbar": "undo redo | formatselect | bold italic backcolor | alignleft aligncenter alignright alignjustify | bullist numlist outdent indent | removeformat | help",
    }
  }
  onEditorChange={[Function]}
  value=""
/>
`;

exports[`TextEditor component tests TextEditor snapshot matches with initial content 1`] = `
<Editor
  cloudChannel="7"
  init={
    {
      "height": 500,
      "menubar": false,
      "plugins": [
        "advlist",
        "autolink",
        "lists",
        "link",
        "image",
        "charmap",
        "print",
        "preview",
        "anchor",
        "searchreplace",
        "visualblocks",
        "code",
        "fullscreen",
        "insertdatetime",
        "media",
        "table",
        "paste",
        "code",
        "help",
        "wordcount",
      ],
      "setup": [Function],
      "toolbar": "undo redo | formatselect | bold italic backcolor | alignleft aligncenter alignright alignjustify | bullist numlist outdent indent | removeformat | help",
    }
  }
  onEditorChange={[Function]}
  value="Initial content"
/>
`;
